### DISCUSSION TITLE
How to set Vec<u8> to body while implementing tonic::codegen::Service

### DISCUSSION THREAD

**Discussion by mustakimali** [hyperium/tonic#815] [Q&A]
I am implementing `tonic::codegen::Service` for my use case to be able to start a mock gRPC server that works with any service at runtime. The response body I need to return is a `Vec<u8>` instead of structs like `HelloReply`.

```rs
// [..]

pub struct MyServer;

impl<B> tonic::codegen::Service<http::Request<B>> for MyServer
where
    B: Body + Send + 'static,
    B::Error: Into<StdError> + Send + 'static,
{
    type Response = http::Response<tonic::body::BoxBody>;
    type Error = Never;
    type Future = tonic::codegen::BoxFuture<Self::Response, Self::Error>;
    // [..]
    fn call(&mut self, req: http::Request<B>) -> Self::Future {
        let mut response : Vec<u8> = Vec::default();
        let body = greeter_code::HelloReply {
            message: "hello".into(),
        }
        .encode(&mut response);

        // now `response` is [u8] representation of `HelloReply`

        Box::pin(async move {
            Ok(http::Response::builder()
                .status(200)
                .header("grpc-status", "0")
                .header("content-type", "application/grpc")
                .body(response) //<-- this is Vec<u8>
                .unwrap())
        })
    }
}
```
This code does not compile as `body` function accepts a different type.

I am trying to figure out how to use that `Vec<u8>` as the parameter of `body` function of `http::Response::builder()`. So far I was able to come up with this

```rs
// [...]
fn call(&mut self, req: http::Request<B>) -> Self::Future {
    return Box::pin(async move {
        // body is the [u8] representation of HelloReply

        let body = prost::bytes::Bytes::from(body);
        let body = http_body::Full::new(body);
        let body =
            http_body::combinators::BoxBody::new(body).map_err(|err| match err {});
        let body = tonic::body::BoxBody::new(body);
        let result = builder.body(body).unwrap();

        Ok(result)
    });
}
// [...]
```

This code compiles, however when I invoke this rpc using the generated client code i get an `Err`.
```
thread 'tests::it_works' panicked at 'called `Result::unwrap()` on an `Err` value: Status { code: Internal, message: "Unexpected EOF decoding stream.", metadata: MetadataMap { headers: {"content-type": "application/grpc", "grpc-status": "0", "date": "Wed, 27 Oct 2021 19:51:12 GMT"} }, source: None }', src/lib.rs:49:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

So to summarise, is it possible to use `Vec<u8>` representation of `HelloReply` and still be able to return a valid gRPC response? Thank you for your help 🙏🏽
